packages/matter.js/src/protocol/interaction/SubscriptionHandler.ts

```diff
@@ -713,7 +713,26 @@ export class SubscriptionHandler {
         );
     }
 
-    async attributeChangeListener<T>(path: AttributePath, schema: TlvSchema<T>, version: number, value: T) {
+    attributeChangeListener<T>(path: AttributePath, schema: TlvSchema<T>, version: number, value: T) {
+        const changeResult = this.attributeChangeHandler(path, schema, version, value);
+        if (MaybePromise.is(changeResult)) {
+            this.attributeUpdatePromises.push(
+                changeResult.then(() => {
```

Comment:
If changeResult can throw then this should probably be `Promise.resolve(changeResult).finally(`